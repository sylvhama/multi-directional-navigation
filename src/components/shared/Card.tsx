import React from "react";
import styled from "styled-components";

import { Base } from "./Base";
import { Props } from "./types";

const StyledBase = styled(Base)`
  position: relative;
  transform: scale(0.7);
  transform-origin: center;
  transition: transform ease 250ms;

  &:last-child:after {
    content: "";
    position: absolute;
    right: -100%;
    width: 100%;
    height: 100%;
  }

  ${({ isFocused }) =>
    isFocused &&
    `
      transform: scale(1);
    `}
`;

export const Card = React.forwardRef<HTMLDivElement, Props>(
  ({ children, isFocused, isPreviousFocus, ...restProps }, ref) => {
    return (
      <StyledBase
        ref={ref}
        tabIndex={isFocused ? 0 : isPreviousFocus ? -1 : undefined}
        isFocused={isFocused}
        {...restProps}
      >
        {children}
      </StyledBase>
    );
  }
);
