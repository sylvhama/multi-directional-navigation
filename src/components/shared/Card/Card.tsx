import React from "react";
import styled from "styled-components";

import { Base } from "components/shared";
import { Props } from "components/shared/types";

const StyledBase = styled(Base)`
  position: relative;
  transform: scale(0.7);
  transform-origin: center;
  transition: transform ease 250ms;

  &:last-child:after {
    content: "";
    position: absolute;
    right: -100%;
    width: 100%;
    height: 100%;
  }

  ${({ isFocused }) =>
    isFocused &&
    `
      transform: scale(1);
    `}
`;

export const Card = React.forwardRef<HTMLDivElement, Props>(
  ({ children, isFocused, ...restProps }, ref) => {
    return (
      <StyledBase ref={ref} isFocused={isFocused} {...restProps}>
        {children}
      </StyledBase>
    );
  }
);
