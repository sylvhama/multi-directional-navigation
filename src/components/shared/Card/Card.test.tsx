import "jest-styled-components";
import React from "react";
import { render } from "@testing-library/react";

import { Card } from "./Card";

describe.each([true, false])("<Card />", (isFocused) => {
  it(`renders style when isFocused is ${isFocused}`, () => {
    const { container } = render(<Card isFocused={isFocused} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
