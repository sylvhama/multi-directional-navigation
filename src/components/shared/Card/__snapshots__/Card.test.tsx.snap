// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders style when isFocused is false 1`] = `
.c0 {
  border: 5px solid #333;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 4rem;
  background-color: #dfe3e8;
  color: #212b36;
}

.c1 {
  position: relative;
  -webkit-transform: scale(0.7);
  -ms-transform: scale(0.7);
  transform: scale(0.7);
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition: -webkit-transform ease 250ms;
  -webkit-transition: transform ease 250ms;
  transition: transform ease 250ms;
}

.c1:last-child:after {
  content: "";
  position: absolute;
  right: -100%;
  width: 100%;
  height: 100%;
}

<article
  class="c0 c1"
/>
`;

exports[`<Card /> renders style when isFocused is true 1`] = `
.c0 {
  border: 5px solid #333;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 4rem;
  background-color: #dfe3e8;
  color: #212b36;
  border-color: #006FBB;
  background-color: #F9FAFB;
  font-weight: bold;
}

.c1 {
  position: relative;
  -webkit-transform: scale(0.7);
  -ms-transform: scale(0.7);
  transform: scale(0.7);
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition: -webkit-transform ease 250ms;
  -webkit-transition: transform ease 250ms;
  transition: transform ease 250ms;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c1:last-child:after {
  content: "";
  position: absolute;
  right: -100%;
  width: 100%;
  height: 100%;
}

<article
  class="c0 c1"
/>
`;
